[Ready for review] etcd-website: Update migration process guidance with etcdctl v3.4

wendy-ha18 opened the pull request:
This PR uses to update the document for migrate etcd v2 to etcd v3 at this issue: https://github.com/etcd-io/website/issues/798

Website review link: https://deploy-preview-1044--etcd.netlify.app/docs/v3.4/how-to-migrate/

- [x] Add prerequisites.
- [x] Define all variables and flags used in the examples in a bulleted list.
- [x] Describe process of migration.
- [x] Add note to guide user that etcdctl migrate command was removed in etcd v3.5.0 ([pull/12971](https://github.com/etcd-io/etcd/pull/12971)). They must use etcdctl v3.4 or earlier to perform the migration ([CHANGELOG-3.5](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#etcdctl-v3-3)). Reference this issue: https://github.com/etcd-io/etcd/issues/14058.
- [x] Included a link explaining the difference between etcd storage format and etcd version.
  - [x] Clarified that the doc guides users on migrating etcd v2store to v3store. All mentions of “v2” and “v3” have been updated to “v2store” and “v3store” for consistency.
  - [x] Added a couple of sentences at the top explaining how v2store relates to etcd v2.X.
  - [x] Added a reference to the official guide for migrating API v2 → API v3 to avoid confusion: https://etcd.io/docs/v3.4/op-guide/v2-migration/.
- [x] Removed the GIF.

wendy-ha18 commented:
Hi @nate-double-u @jberkus @ronaldngounou, this PR is ready to review. I'm a bit hesitate to backport to v3.7 cause I want to get feedbacks for it first. Please help me revise the migration process here. Thanks everyone!

nate-double-u commented:
I think there are a couple paths here:
1. if we don't want to support migration from v2 to v3.5 and above, then we should remove migration from v3.5, v3.6, and v3.7.
2. if we do want to keep supporting migration then it looks like the path is to go from v2 to v3.4, then do a stepwise upgrade from there (v3.4 to v3.5, v3.5 to v3.6 and so on). I don't think think this is a great user experience, and would rather go with option 1.

jberkus commented on content/en/docs/v3.6/tasks/operator/how-to-migrate.md:
```diff
@@ -4,10 +4,73 @@ description: etcd v2 to v3 migration guide
 ---
 
-`migrate` to transform etcd v2 to v3 data:
+`migrate` to transform etcd v2 data model to v3 data model.
```
```suggestion
`migrate` to transform etcd v2 data format to v3 data format.
```

"model" generally refers to the schema, rather than the storage format

nate-double-u commented:
We should test that this migration path works still. Stepwise upgrades may be the only way to go.

wendy-ha18 commented:
Q: Should we keep the Visual guide GIF here? this's quite hard to maintain and I'm not sure about the accuracy of it now as it was created 3 years ago.

![12_etcdctl_migrate_2016061602](https://storage.googleapis.com/etcd/demo/12_etcdctl_migrate_2016061602.gif)

ivanvc commented:
> Q: Should we keep the Visual guide GIF here? this's quite hard to maintain and I'm not sure about the accuracy of it now as it was created 3 years ago.

I spoke about the GIFs in the tutorial pages with James before. I agree that the problem is that they can get outdated, and there's no easy way to update/reproduce them. An alternative could be to use a tool like [VHS](https://github.com/charmbracelet/vhs), write a script for each GIF, and use our official Docker images to reproduce the same GIFs later on.

jberkus commented:
Two more things per Ivan's comments.

- Most mentions of "v2" should actually say "v2store".  A lot of mentions of v3 should also say v3store.
- There should be a couple sentences at the top that explain how v2store relates to etcd v2.X

Also: mention that there's also a change of API, and link to this document: https://etcd.io/docs/v3.4/op-guide/v2-migration/

jberkus commented:
We should discuss `v2` vs `v2store` before merging this.  Also whether it's "v2 store" or "v2store".  Once we discuss that, we can merge.

jberkus commented:
@wendy-ha18 this looks good but it's still marked as a draft.  Are you ready to merge it?

wendy-ha18 commented:
Hi @jberkus and everyone, this PR is ready for final review now.

I included all of these changes:
Website review link: https://deploy-preview-1044--etcd.netlify.app/docs/v3.4/how-to-migrate/

- [x] Add prerequisites.
- [x] Define all variables and flags used in the examples in a bulleted list.
- [x] Describe process of migration.
- [x] Add note to guide user that etcdctl migrate command was removed in etcd v3.5.0 ([pull/12971](https://github.com/etcd-io/etcd/pull/12971)). They must use etcdctl v3.4 or earlier to perform the migration ([CHANGELOG-3.5](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#etcdctl-v3-3)). Reference this issue: https://github.com/etcd-io/etcd/issues/14058.
- [x] Included a link explaining the difference between etcd storage format and etcd version.
  - [x] Clarified that the doc guides users on migrating etcd v2store to v3store. All mentions of “v2” and “v3” have been updated to “v2store” and “v3store” for consistency.
  - [x] Added a couple of sentences at the top explaining how v2store relates to etcd v2.X.
  - [x] Added a reference to the official guide for migrating API v2 → API v3 to avoid confusion: https://etcd.io/docs/v3.4/op-guide/v2-migration/.
- [x] Removed the GIF.

jberkus commented:
Can you fix the linter failure?
